rst_files/linear_algebra.rst

```diff
@@ -992,10 +992,12 @@ As expected, the image :math:`Av` of each :math:`v` is just a scaled version of
     A = [1 2
          2 1]
     evals, evecs = eigen(A)
+    
     a1, a2 = evals
     evecs = selectdim.(Ref(evecs), 1, 1:2)
     eig_1 = reduce(hcat, ([0, v] for v in first.(evecs)))
```

Comment:
I think this pattern of a reduction on `hcat` is **way** to difficult to understand for new students.  It took me a while to figure it out.

We have never even taught them that `hcat` is a function, that reduce is a higher order function, etc. 

The old for loops may be easier to understand.

Reply:
The for loops were pretty hard to understand as well. Some alternatives are
```
reduce(hcat, vcat.(0, first.(evecs)))
```
```
eig_1 = [zero(n, 1), first.(evecs)']
```

Reply:
```
eig_1 = [zero(n, 1), first.(evecs)']
```
That is still pretty tricky.  Do we really need to use broadcasting here?  This is one of the first lectures people are going to read after just barely learning julia.  How big is the vector?  Can we just concatenate it with a hardcoded number of columns?